pkg/nad/masterplugin.go

```diff
@@ -393,3 +393,126 @@ func (plugin *MasterIPVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error)
 
 	return plugin.masterPlugin, nil
 }
+
+// MasterBondPlugin provides struct for MasterPlugin to create a NAD for bonded interfaces.
+type MasterBondPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterBondPlugin creates new instance of MasterBondPlugin.
+func NewMasterBondPlugin(name, mode string) *MasterBondPlugin {
+	validModes := map[string]bool{
+		"balance-rr":    true,
+		"active-backup": true,
+		"balance-xor":   true,
+	}
+
+	glog.V(100).Infof(
+		"Initializing new MasterBondPlugin structure %s", name)
+
+	builder := MasterBondPlugin{
+		masterPlugin: &MasterPlugin{
+			CniVersion: "0.3.1",
+			Name:       name,
+			Type:       "bond",
+			Mode:       mode,
+		},
+	}
+
+	// Check if the provided mode is valid
+	if !validModes[mode] {
+		glog.V(100).Infof("error: invalid mode type %s used for MasterBondPlugin bond interface", mode)
+
+		builder.errorMsg = "Bond mode type is not valid"
+	}
+
+	if builder.masterPlugin.Name == "" {
+		glog.V(100).Infof("error NewMasterBondPlugin name can not be empty")
+
+		builder.errorMsg = "NewMasterBondPlugin name is empty"
+	}
+
+	return &builder
+}
+
+// WithLinksInContainer defines linksInContainer configuration to MasterBondPlugin.
+func (plugin *MasterBondPlugin) WithLinksInContainer(linksInContainer bool) *MasterBondPlugin {
+	plugin.masterPlugin.LinksInContainer = linksInContainer
```

Comment:
Please add log:
glog.V(100).Infof("Adding LinksInContainer %d to MasterBoundPlugin %s", linksInContainer, plugin.masterPlugin.Name)